components/script/dom/mutationrecord.rs

```diff
@@ -0,0 +1,135 @@
+use core::ptr;
+use dom_struct::dom_struct;
+use dom::bindings::codegen::Bindings::MutationRecordBinding::MutationRecordBinding::MutationRecordMethods;
+use dom::bindings::js::{JS, Root};
+use dom::bindings::str::DOMString;
+use dom::bindings::reflector::Reflector;
+use dom::node::Node;
+use dom::nodelist::NodeList;
+use dom::window::Window;
+use std::default::Default;
+
+#[dom_struct]
+pub struct MutationRecord {
+  reflector_: Reflector,
+
+//  readonly attribute DOMString record_type;
+  record_type: DOMString,
+
+//  [SameObject]
+//  readonly attribute Node target;
+  target: Root<Node>,
```

Comment:
Hi Josh, when I make this change, it gives me the following error:

`error[E0053]: method `Target` has an incompatible type for trait
   --> /home/sumit/Dropbox/NCSU/OODD/Servo/MyWork/servo/components/script/dom/mutationrecord.rs:38:25
    |
38  |     fn Target(&self) -> JS<Node> {
    |                         ^^^^^^^^ expected struct `dom::bindings::js::Root`, found struct `dom::bindings::js::JS`
    | 
   ::: /home/sumit/Dropbox/NCSU/OODD/Servo/MyWork/servo/target/debug/build/script-bb44ff0630eb7b72/out/Bindings/MutationRecordBinding.rs
    |
617 |     fn Target(&self) -> Root<Node>;
    |                         ---------- type in trait
    |
    = note: expected type `fn(&dom::mutationrecord::MutationRecord) -> dom::bindings::js::Root<dom::node::Node>`
               found type `fn(&dom::mutationrecord::MutationRecord) -> dom::bindings::js::JS<dom::node::Node>`
`